### DISCUSSION TITLE
how to render external data (csv or yaml) with a shortcode

### DISCUSSION THREAD

**Discussion by StefanoCecere** [squidfunk/mkdocs-material#4700] [Q&A]
Hi. i come from Hugo, where it is very easy to write a **shortcode** (a function) that gets called from the .md pages passing variables and renders back the html/md.

for example i have a CSV with 100 records with a country field and i want to render them in different pages, like < render_list("italy") %> or <% render_list(spain) %>

so to keep data separated from the pages, and a function for everything.

the only solution that gets near to this is the [macros](https://mkdocs-macros-plugin.readthedocs.io/en/latest/) plugin, right?
but it isn't so easy to write a custom function.. i have to install modules, etc...

isn't there a more direct or easier solution?
is a better _mkdocs_apidoc_ the solution?

thank you

maybe this question is related to MkDocs, but i decided to use it just because of this Material theme

**Comment by squidfunk** [comment]
This question is definitely related to MkDocs, but maybe some users can share their workflows.
